Makefile

```diff
@@ -1,21 +1,26 @@
-#
-# (C) Copyright IBM Corp. 2022.
+# (C) Copyright Confidential Containers Contributors
 # SPDX-License-Identifier: Apache-2.0
-#
 
 .PHONY: all build check fmt vet clean image deploy delete
-ifndef CLOUD_PROVIDER
-$(error CLOUD_PROVIDER is not set)
-endif
+
 
 ARCH        ?= $(subst x86_64,amd64,$(shell uname -m))
+# Add providers that needs to be included in the builds
+# Default list doesn't include libvirt
+BUILD_PROVIDERS ?= aws,azure,ibmcloud,vsphere
+CLOUD_PROVIDER ?=
 GOOPTIONS   ?= GOOS=linux GOARCH=$(ARCH)
-GOFLAGS     ?= -tags=$(CLOUD_PROVIDER)
+GOFLAGS     ?= -tags=$(BUILD_PROVIDERS)
 BINARIES    := cloud-api-adaptor agent-protocol-forwarder
 SOURCEDIRS  := ./cmd ./pkg
 PACKAGES    := $(shell go list $(addsuffix /...,$(SOURCEDIRS)))
 SOURCES     := $(shell find $(SOURCEDIRS) -name '*.go' -print)
 
+ifeq (,$(findstring libvirt,$(BUILD_PROVIDERS)))
+	# Force static build since libvirt provider is not part of the providers to be built. 
+	override GOOPTIONS += CGO_ENABLED=0
+endif
+
```

Comment:
I think this will allow to revert the caa, forwader targets below back to:
```
$(BINARIES): .git-commit $(SOURCES)
        $(GOOPTIONS) go build $(GOFLAGS) -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=$(shell echo "unknown")' -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=$(shell cat .git-commit)'" -o "$@" "cmd/$@/main.go"
```

Reply:
i just noticed it will not take into account in some of the targets below such as vet, escapse if the default is without cgo should they also used without? IDK

Reply:
I ran the following targets without any issues. 
```
make vet
make escapes
BUILD_PROVIDERS="aws,azure,ibmcloud,vsphere,libvirt" make vet
BUILD_PROVIDERS="aws,azure,ibmcloud,vsphere,libvirt" make escapes
```

Reply:
notice that it will run with CGO regardless to if libivirt i specified or not, I'm not sure if it's a problem or not, good that it passes..

Reply:
I think it shouldn't be a problem since these are static checks. We can revisit if we see any issues.